test/integration/hds_integration_test.cc

```diff
@@ -496,15 +496,41 @@ TEST_P(HdsIntegrationTest, TwoEndpointsSameLocality) {
 
   // Receive updates until the one we expect arrives
   ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
-  while (!(checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(0),
+  while (!(checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(0)
+                                           .locality_endpoints_health(0)
+                                           .endpoints_health(0),
                                        envoy::config::core::v3::UNHEALTHY,
                                        host_upstream_->localAddress()) &&
-           checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(1),
+           checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(0)
+                                           .locality_endpoints_health(0)
```

Comment:
although you might have to restructure the while-loop a little, I think this test would be much more readable if you captured `response_.endpoint_health_response().cluster_endpoints_health(0).locality_endpoints_health(0)` in a temp. It's used many times below.

Reply:
Thanks for the suggestion. I agree that the length of these statements got a little bit out of hand. I changed the check in the while loop to be much shorter, and use temp vars along the integration test now to make it much more readable.